### PR TITLE
feat: warn when advancing into match with no starting XI

### PR DESCRIPTION
**Pull request by trask-onvx**
## Summary
- Adds a warning modal when the player clicks **Advance Week** (and the next calendar entry is a match) or **Play Match** with an empty starting XI
- Modal displays "You haven't selected your starting lineup! Go to Squad to assign players." with two buttons:
  - **Go to Squad** — navigates to the squad page
  - **Play Anyway** — dismisses the warning and continues
- Modal lives in the Dashboard component using existing MODAL/BTN token patterns (amber border, matches holiday confirm modal style)

## Files modified
- `src/App.jsx` — added `showLineupWarning` state, intercept on Advance Week and Play Match button clicks, pass props to Dashboard
- `src/components/ui/Dashboard.jsx` — imported MODAL token, added warning modal rendering with new props

## How to test
1. Start a new game (any mode)
2. Do NOT assign any starting XI
3. Click **Advance Week** — if the next calendar entry is a match, the warning modal should appear
4. Test **Go to Squad** button navigates to squad page
5. Test **Play Anyway** button dismisses modal and advances the week
6. Also test: when a match is pending (PLAY MATCH button showing), clicking it with empty XI shows the warning
7. Verify on mobile viewport (≤768px) — modal should be responsive

## Build
⚠️ Build not verified in VM (npm registry unreachable from Cowork) — please run `npx --no vite build --mode development` locally before merging.

🤖 Generated with Claude Cowork

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -6830,6 +6844,18 @@ function FootballManager() {
         ultimatumTarget={ultimatumTarget}
         ultimatumGamesLeft={ultimatumGamesLeft}
         gameMode={gameMode}
+        showLineupWarning={showLineupWarning}
+        onDismissLineupWarning={() => setShowLineupWarning(false)}
+        onLineupWarningGoToSquad={() => {
+          setShowLineupWarning(false);
```
Comment (calo-onvx) [inline src/App.jsx]: **Bug: "Play Anyway" takes the wrong path when triggered from the Play Match button.**

In plain English: there are two buttons that can trigger the lineup warning:
- **Advance Week** — its normal action is just `advanceWeek()`
- **Play Match** — its normal action has extra logic before advancing (e.g. checking the Altitude Trials modifier for minimum ATK players)

Right now, `onLineupWarningPlayAnyway` always calls `advanceWeek()`. That's correct when the warning came from Advance Week. But when it came from **Play Match**, it skips all the match-button checks (like the `minAtkPlayers` modifier check at line ~4285).

**Fix:** Track *which* button triggered the warning, then "Play Anyway" should resume that button's original flow. One approach: store the origin (`"advance"` vs `"match"`) in state alongside `showLineupWarning`, then branch in the Play Anyway handler.

File: src/components/ui/Dashboard.jsx [hunk 2]
```diff
@@ -837,6 +838,64 @@ export function Dashboard({
...
       </div>{/* end newspaper frame */}
 
+      {/* ═══ LINEUP WARNING MODAL ═══ */}
+      {showLineupWarning && (
+        <div style={{ ...MODAL.backdrop, zIndex: Z.confirm }}>
+          <div style={{
+            ...MODAL.box,
+            border: `3px solid ${C.amber}`,
+            padding: mob ? "30px 20px" : "40px 44px",
+            width: mob ? "90%" : "auto",
```
Comment (calo-onvx) [inline src/components/ui/Dashboard.jsx]: **Bug: `C.amber` is not a defined token — this will render as `undefined`.**

In plain English: the project's color tokens are defined in `src/data/tokens.js` under the `C` object. There's `C.green`, `C.red`, `C.text`, `C.textMuted`, etc. — but no `C.amber`. So `C.amber` evaluates to `undefined`, meaning this border and the text colors using it will silently fail (no border shown, default text color).

**Fix:** Either:
- Use an inline hex value: `"#fbbf24"` (Tailwind amber-400) — define it once at the top of the modal block as a local const
- Or add `amber: "#fbbf24"` to the `C` object in `src/data/tokens.js` (but check with the project lead first since CONTRIBUTING.md says not to add new tokens without reason)

File: src/App.jsx [hunk 3]
```diff
@@ -6830,6 +6844,29 @@ function FootballManager() {
...
+        onLineupWarningGoToSquad={() => {
+          setShowLineupWarning(null);
+          setShowAchievements(false); setShowTable(false); setShowCalendar(false);
+          setShowCup(false); setShowTransfers(false); setShowLegends(false);
+          setShowSquad(true);
+        }}
+        onLineupWarningPlayAnyway={() => {
+          const origin = showLineupWarning;
+          setShowLineupWarning(null);
+          if (origin === "match") {
```
Comment (calo-onvx) [inline src/App.jsx]: **Bug: `playMatchBtnRef` is not defined anywhere in the codebase.**

The idea of re-clicking the Play Match button to replay its full onClick flow is creative, but:
1. There's no `playMatchBtnRef` declared as a `useRef` in App.jsx
2. Even if you created one and attached it, re-clicking the button would hit the empty startingXI check *again* at line 4283 and loop back to showing the warning — infinite loop.

**Simpler fix:** Don't re-trigger the button. Instead, extract the match-button's post-warning logic into a function and call it directly. Or, since the only extra check in the match path right now is the `minAtkPlayers` Altitude Trials check (which doesn't apply with an empty XI anyway), you could just call `advanceWeek()` for both paths. The match-button checks are all about *validating the lineup*, which is moot when the user chose "Play Anyway" with no lineup.

So this could simplify to:
```js
onLineupWarningPlayAnyway={() => {
  setShowLineupWarning(null);
  advanceWeek();
}}
```
The origin tracking is still good to have for future match-path checks, but right now both paths can safely call `advanceWeek()`.

File: src/components/ui/Dashboard.jsx (hunk 2 above)
Comment (calo-onvx) [inline src/components/ui/Dashboard.jsx]: **Still unresolved from round 1: `C.amber` is not a defined token.**

This border and the title color on line 856 both reference `C.amber`, which doesn't exist in `src/data/tokens.js`. They'll render as `undefined`.

**Fix:** Replace `C.amber` with a local const:
```js
const AMBER = "#fbbf24";
```
Then use `AMBER` in place of `C.amber` throughout this modal block.

**Comment by calo-onvx** [comment]
**Correction — Calo**

My mistake on `C.amber` — it does exist in `src/data/tokens.js` (line 17). That comment is resolved, no change needed.

The `playMatchBtnRef` issue from round 2 still stands though.

**Comment by calo-onvx** [comment]
Build verified — passes clean.

```
✓ 1057 modules transformed.
dist/index-CFtWdsCH.js   1,290.53 kB │ gzip: 367.43 kB
✓ built in 1.85s
```

Chunk size warning is expected (pre-existing). No errors. Good to merge.
